client/src/routes/carbonsensor/carbon.js

```diff
@@ -0,0 +1,252 @@
+import moment from 'moment';
+import { connect } from 'react-redux';
+import { Spinner, Alert } from 'react-bootstrap';
+import { Line, defaults } from 'react-chartjs-2';
+import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';
+import sliderStyle from 'rc-slider/assets/index.css';
+import Slider from 'rc-slider';
+import getChartData from './chart-dataset';
+import Icon from './carbonIcon/carbon-icon';
+import Loader from '../../components/Loader/Loader';
+import { setFilter, changeCarbonStatus } from '../../actions/carbonmonoxide';
+
+defaults.global.defaultFontFamily = 'Montserrat';
+
+const options = {
+  defaultSortName: 'EventID',
+  defaultSortOrder: 'desc',
+  display: { maintainAspectRatio: true },
+  legend: { display: false },
+};
+
+const sliderOption = {
+  borderColor: 'grey',
+};
+
+const marks = {
+  0: {
+    label: <strong>Last Hour</strong>,
+    style: {
+      color: 'red',
+    },
+  },
+  1: <strong>Day</strong>,
+  2: <strong>Week</strong>,
+  3: <strong>Month</strong>,
+  4: {
+    label: <strong>Year</strong>,
+  },
+};
+
+class CarbonMonoxideSensor extends Component {
+  static propTypes = {
+    dispatchChangeStatus: PropTypes.func.isRequired,
+    dispatchSetFilter: PropTypes.func.isRequired,
+    error: PropTypes.string.isRequired,
+    events: PropTypes.arrayOf(
+      PropTypes.shape({
+        Created: PropTypes.string.isRequired,
+        CreatedAt: PropTypes.string.isRequired,
+        EventID: PropTypes.number.isRequired,
+        ID: PropTypes.number.isRequired,
+        UpdatedAt: PropTypes.string.isRequired,
+        device_type: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+        signal: PropTypes.number.isRequired,
+      }),
+    ).isRequired,
+    info: PropTypes.shape({
+      Name: PropTypes.string.isRequired,
+      Status: PropTypes.number.isRequired,
+      Type: PropTypes.string.isRequired,
+    }).isRequired,
+    isLoading: PropTypes.bool.isRequired,
+    removeInterval: PropTypes.func.isRequired,
+  };
+
+  componentWillUnmount() {
+    const { removeInterval } = this.props;
+    removeInterval();
+  }
+
+  handleOnClick = status => {
+    const { dispatchChangeStatus } = this.props;
+    if (status === 1) {
+      dispatchChangeStatus(false);
+    } else {
+      dispatchChangeStatus(true);
+    }
+  };
+
+  handleOnSliderChange = value => {
+    switch (value) {
+      case 1:
+        this.getFilterData('days', 1);
+        break;
+      case 2:
+        this.getFilterData('weeks', 1);
+        break;
+      // eslint-disable-next-line no-magic-numbers
+      case 3:
+        this.getFilterData('months', 1);
+        break;
+      // eslint-disable-next-line no-magic-numbers
+      case 4:
+        this.getFilterData('years', 2);
+        break;
+      default:
+        this.getFilterData('hours', 1);
+    }
+  };
+
+  render() {
+    const { events, info, isLoading, error } = this.props;
+
+    if (error) {
```

Comment:
So if you have error, your chart disappears? Fix it please, make error appear above or below the chart